ui/src/components/facets/HistogramFacet.js

```diff
@@ -177,6 +185,52 @@ class HistogramFacet extends Component {
       })
     );
 
+    // vega-lite spec is easier to construct than vega spec. But certain
+    // properties aren't available in vega-lite spec (vega-lite is a subset of
+    // vega). So construct vega-lite spec, compile to vega spec, edit properties
+    // that are only available in vega, then render Vega component.
+    vegaLiteSpec.data = data;
+    const vegaSpec = vl.compile(vegaLiteSpec).spec;
+
+    // Setting align removes whitespace over top bar.
+    // When https://github.com/vega/vega-lite/issues/4741 is fixed, set align
+    // normally.
+    vegaSpec.scales[1].align = 0;
+
+    // Add tooltips to axis labels. vega-lite spec doesn't have "encode".
+    let facetValueIndexStr = "";
+    if (isCategorical(this.props.facet)) {
+      // See https://github.com/vega/vega/issues/1719 for why we need round().
+      // Remove isNaN when we're using a vega version with
+      // https://github.com/vega/vega/pull/1720
+      facetValueIndexStr =
+        "round((isNaN(datum.index)?0:datum.index) * (length(data('source_0'))/2 - 1))";
```

Comment:
As we've discussed, lets add comments to explain why we need to do this. Its pretty clever, but really unclear.

Reply:
Done